backend/fms_core/management/commands/_delete_individual.py

```diff
@@ -0,0 +1,86 @@
+from django.apps import apps
+import reversion
+import json
+import logging
+from reversion.models import Version
+from fms_core.models import Individual
+
+# Parameters required for this curation
+ACTION = "action"                     # = delete_individual
+CURATION_INDEX = "curation_index"     # Number indicating the order in which this action was performed during the curation.
+COMMENT = "comment"                   # An optional comment to be stored in the logs
+INDIVIDUAL_NAMES = "individual_names" # An array of individual name that are to be deleted.
+USER_ID = "requester_user_id"         # The user id of the person requesting the curation. Optional. If left empty, uses biobankadmin id.
+
+# Curation params template
+# { CURATION_INDEX: 1,
+#   ACTION: "delete_individual",
+#   COMMENT: "Dr. No asked to delete these individuals that are not used.",
+#   INDIVIDUAL_NAMES: ["Roger", "TiPaul", "Gertrude"], # List of individual names (unique) to be deleted
+#   USER_ID: 5
+# }
+
+# This curation deletes only individual that are not referenced. Remove individual from samples references using "update_field_value" first.
+# To delete parents, a first curation need to be run to delete the child individual (or an update), then parents will be unreferenced.
+
+# function that checks the references to an individual and list them.
+def references(individual):
+    links = []
+    children = list(individual.mother_of.all())
+    if children:
+        links.append(f"Mother of {[child.name for child in children]}.")
+        children = None
+    children = list(individual.father_of.all())
+    if children:
+        links.append(f"Father of {[child.name for child in children]}.")
+        children = None
+    samples = list(individual.samples.all())
+    if samples:
+        links.append(f"Has samples {[sample.id for sample in samples]}")
+    return links
+
+
+def delete_individual(params, objects_to_delete, log):
+    log.info("Action [" + str(params[CURATION_INDEX]) + "] Update Field Value started.")
+    log.info("Comment [" + str(params.get(COMMENT, "None")) + "].")
+    log.info("Individual names : " + str(params[INDIVIDUAL_NAMES]) + ".")
+    log.info("Requester id : " + str(params.get(USER_ID)))
+
+    # initialize the curation
+    curation_code = params.get(CURATION_INDEX, "Invalid index")
+    error_found = False
+    name_array = params[INDIVIDUAL_NAMES]
+    user_id = params.get(USER_ID)
+
+    try:
+        individual_model = apps.get_model("fms_core", "Individual")
+        count_deletes = 0
```

Comment:
count_deleted ?

Reply:
potato potéto to me ;) i'll change it...

Reply:
you were the one who asked me to change DELETE TO DELETED in a comment message :P same should go for variable names ;P

Reply:
Eheh this was English syntax correction which have much more clear cut rules than variable naming ;) But i agree that it was of very modest importance ;).

Reply:
Code is self-documenting :P variable names ARE in some way important hehe